checker/src/org/eclipse/jgit/annotations/NonNull.java

```diff
@@ -0,0 +1,12 @@
+package org.eclipse.jgit.annotations;
+
+import java.lang.annotation.*;
+
+import org.checkerframework.framework.qual.TypeQualifier;
+
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE})
```

Comment:
In the Checker Framework's copy of the annotation, this line should be `@Target({ElementType.TYPE_USE})` (even though what you have written is what is in jgit's copy).
Likewise in Nullable.java.
Other than that, the pull request looks good to me.